src/core/pattern.js

```diff
@@ -956,13 +956,20 @@ class MeshShading extends BaseShading {
   }
 
   getIR() {
+    const { bounds } = this;
+    // Ensure that the shading has non-zero width and height, to prevent errors
+    // in `pattern_helper.js` (fixes issue17848.pdf).
+    if (bounds[2] - bounds[0] === 0 || bounds[3] - bounds[1] === 0) {
+      throw new FormatError(`Invalid MeshShading bounds: [${bounds}].`);
```

Comment:
It's useless to draw an empty shading but why is it an error ?

Reply:
Since we didn't previously handle errors properly during /Shading parsing, which this patch also adds, it seemed overall easier (implementation wise) to treat this case as an error as well.
And in some ways I think this particular case *can* be thought of as an error too, since the "empty" /Shading shouldn't have been included in the PDF document in the first place.